api/envoy/v6/http/path_matcher/config.proto

```diff
@@ -26,28 +26,20 @@ message PathMatcherRule {
   // Operation name (selector) used as a unique ID for the operation.
   string operation = 2 [(validate.rules).string.min_bytes = 1];
 
-  // If true, the path parameters in the `pattern` must be extracted.
+  // If set, the path parameters in the `pattern` must be extracted.
   // Ref:
   // https://cloud.google.com/endpoints/docs/openapi/openapi-extensions#understanding_path_translation
-  bool extract_path_parameters = 3;
+  PathParameterExtractionRule path_parameter_extraction = 3;
 }
 
-// FieldName stores the snake name to JSON name mapping as specified in
-// Service config.
-// For example, for a URI template: /{foo_bar}
-//  * snake_name = "foo_bar"
-//  * json_name = "fooBar"
-message SegmentName {
-  // Snake name of a segment.
-  string snake_name = 1;
-  // JSON name of a segment.
-  string json_name = 2;
+message PathParameterExtractionRule {
```

Comment:
why do we need an extra Message type?   Can we just add a map to PathMatcherRule message

Reply:
By reading c++ code, I understand now why you need a separate message. It is fine.

Reply:
Prefer not to wrapper with message

Reply:
We use the nullability of the message to signify whether path param extraction is needed.
- No message = no extraction
- Set message = needs extraction
- Set message with mappings = needs extraction and mapping
I'll clarify in the comment.

Reply:
I see your way of differentiating no/needs extraction. 

Maybe still add a bool since using nullability of the message is not straightforward and easy to miss assigning an empty map on go code side. WDYT.

Reply:
I prefer the current approach. Otherwise, you introduce a 4th case:
- Boolean is set to false, but map has mappings.

What do we do in this case? Reject the config? Ignore the mappings but accept the config?

It would be easier to just use message nullability to ensure this case will never happen.